From 4b5238d5a7422f71abc4dbe8a7f01ab54dc9599b Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Tue, 11 Apr 2017 16:42:01 +0200
Subject: [PATCH 12/14] start: fix error handling when limits fail to apply

(The code was moved here from the child side of the startup
without adapting the error case.)

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 src/lxc/start.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/start.c b/src/lxc/start.c
index 29edb8f7..8c2d2182 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -1246,7 +1246,7 @@ static int lxc_spawn(struct lxc_handler *handler)
 
 	if (!lxc_list_empty(&handler->conf->limits) && setup_resource_limits(&handler->conf->limits, handler->pid)) {
 		ERROR("failed to setup resource limits for '%s'", name);
-		return -1;
+		goto out_delete_net;
 	}
 
 	if (!cgroup_setup_limits(handler, true)) {
-- 
2.11.0

